AirCasting/Settings/SettingsView.swift

```diff
@@ -82,71 +74,74 @@ struct SettingsView: View {
                 }
             }
         }.sheet(isPresented: $showModal, content: {
-            BackendSettingsModalView()
+            BackendSettingsModalView(urlProvider: urlProvider)
         })
     }
+}
+
+private struct BackendSettingsModalView: View {
```

Comment:
IMO it would be better to have this in a separate file (but not a strong opinion)